profiles/common/modules/features/nexteuropa_trackedchanges/nexteuropa_trackedchanges.admin.inc

```diff
@@ -0,0 +1,700 @@
+      drupal_set_message($message, 'error');
+    }
+    else {
+      $featureset = feature_set_get_featureset('nexteuropa_trackedchanges');
+      feature_set_disable_feature_set($featureset);
+      // Required to flush thecache to ensure that all references to the
+      // module have disappeared.
+      // @see system_modules_submit()
+      drupal_flush_all_caches();
+    }
+  }
+  else {
+    // An error occurred.
+    // $operations contains the operations that remained unprocessed.
+    $error_operation = reset($operations);
+    $message_args = array(
+      '%error_operation' => $error_operation[0],
+      '@arguments' => print_r($error_operation[1], TRUE),
+    );
+    $message = t('An error occurred during the deactivation on the 
+      %error_operation with arguments: @arguments', $message_args);
+    $message .= '<br />';
+    $message .= t('The deactivation is impossible.');
+    drupal_set_message($message, 'error');
+  }
+}
+
+
+/**
+ * Finished callback of the batch field for tracked changes values.
+ *
+ * It is run in the case where the feature is disabled.
+ *
+ * @see _nexteuropa_trackedchanges_feature_disable_check_submit()
+ */
+function _nexteuropa_trackedchanges_feature_disable_field_scanning_finished($success, $results, $operations) {
+  if ($success) {
+    // Here we do something meaningful with the results.
+    if (!empty($results['detected_entities'])) {
+      $message = t('The deactivation stopped because tracked changes have been 
+        detected in contents.');
+      $message .= '<br />';
+      $message .= t('Please accept or reject them before proceeding to the deactivation; 
+      the list of entities with tracked changes is available !here.',
+        array('!here' => l(t('here'), 'admin/content/tracked_changes')));
+      drupal_set_message($message, 'error');
+    }
+    else {
+      module_disable(array('nexteuropa_trackedchanges'));
+      // Required to flush thecache to ensure that all references to the
+      // module have disappeared.
+      // @see system_modules_submit()
+      drupal_flush_all_caches();
+    }
+  }
+  else {
+    // An error occurred.
+    // $operations contains the operations that remained unprocessed.
+    $error_operation = reset($operations);
+    $message_args = array(
+      '%error_operation' => $error_operation[0],
+      '@arguments' => print_r($error_operation[1], TRUE),
+    );
+    $message = t('An error occurred during the deactivation on the 
+      %error_operation with arguments: @arguments', $message_args);
+    $message .= '<br />';
+    $message .= t('The deactivation is impossible.');
+    drupal_set_message($message, 'error');
+  }
+}
+
+/**
+ * Finished callback of the batch field for tracked changes values.
+ *
+ * It is run in the case where the module is disabled.
+ *
+ * @see _nexteuropa_trackedchanges_system_disable_check_submit
+ */
+function _nexteuropa_trackedchanges_system_disable_field_scanning_finished($success, $results, $operations) {
```

Comment:
For a better readability, wouldn'it be possible to reduce the number of 'else' ?
Ex: by assigning a default value to $message and use only once the function drupal_set_message() ?